### PR TITLE
TMC SPI Daisy chain support

### PR DESCRIPTION
**Pull request by MitchBradley**
For daisy-chained SPI:
1) The first, and only the first, TMC in the config file must have
cs_pin
2) Each TMC must have spi_index
3) The spi_index values must be unique

### CODE REVIEW

**Comment by boucz** [comment]
Hi guys,
thanks for working on daisy chain TMC !
I'm using your design @bdring (based on  4_Axis_SPI_CNC) with few changes on spindle, probe and endstops.
I use 3 TMC2130.
My setup is working with grbl-esp32 in daisy chain mode.
When I try your commit "WIP Testing", I have this error. I did try modding the config file without success.
image

Can I do anything to help?

**Comment by bdring** [comment]
It does not work yet

**Comment by boucz** [comment]
> 
> 
> It does not work yet

Ok.
If you need beta tester ;)

**Comment by MitchBradley** [comment]
Bart has a test case and is working on it so I expect that it will be solved soon.

**Comment by bdring** [comment]
@boucz Try it now. It might be working. I am having trouble with my hardware

File: FluidNC/src/Motors/TrinamicSpiDriver.cpp [hunk 1]
```diff
@@ -34,7 +34,8 @@ namespace MotorDrivers {
         }
 
         if (_driver_part_number == 2130) {
-            tmcstepper = new TMC2130Stepper(cs_id, _r_sense, -1);  // TODO hardwired to non daisy chain index
+            //log_info("ID: " << cs_id << " index:" << _spi_index);
+            tmcstepper = new TMC2130Stepper(cs_id, _r_sense, _spi_index);  // TODO hardwired to non daisy chain index
```
Comment (MitchBradley) [inline FluidNC/src/Motors/TrinamicSpiDriver.cpp:38]: I guess the TODO can go

Comment (boucz) [reply FluidNC/src/Motors/TrinamicSpiDriver.cpp:38]: not working but there is a change, Z motor passed the test :
image

**Comment by bdring** [comment]
As I recall now, all axes need to be set up before you talk to them, because it needs to know how many drivers there are to shift the responce through.

**Comment by bdring** [comment]
I think I got it working. Tests are done after all motor have been set up.

I need to step away for a bit. I'll clean it up and push the changes to the PR later.

```
[MSG:INFO: Axis count 4]
[MSG:INFO: Axis X (150.000,450.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     Trinamic TMC2130 Step:gpio.12 Dir:gpio.14 CS:gpio.17 Disable:NO_PIN Index:1 R:0.110]     
[MSG:INFO: Axis Y (150.000,450.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     Trinamic TMC2130 Step:gpio.27 Dir:gpio.26 CS:NO_PIN Disable:NO_PIN Index:2 R:0.110]      
[MSG:INFO: Axis Z (150.000,450.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     Trinamic TMC2130 Step:gpio.15 Dir:gpio.2 CS:NO_PIN Disable:NO_PIN Index:3 R:0.110]       
[MSG:INFO: Axis A (150.000,450.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     Trinamic TMC2130 Step:gpio.33 Dir:gpio.32 CS:NO_PIN Disable:NO_PIN Index:4 R:0.110]      
[MSG:INFO: Trinamic driver test passed]
[MSG:INFO: Trinamic driver test passed]
[MSG:INFO: Trinamic driver test passed]
[MSG:INFO: Trinamic driver test passed]
[MSG:INFO: No spindle]
[MSG:INFO: Using spindle NoSpindle]
```

**Comment by boucz** [comment]
seems to work :)
initialization ok, movements ok.
Will try a few paths when I have some time !
great work, thanks !

**Comment by boucz** [comment]
I've been able to run 2 gcode without issue :)
Thanks !

**Pull request merged**